pkg/builder/s2i/s2i.go

```diff
@@ -21,12 +21,34 @@ import (
 	"github.com/apache/camel-k/pkg/builder"
 )
 
+func init() {
+	builder.RegisterSteps(Steps)
+}
+
+type steps struct {
+	Publisher   builder.Step
+	ReplaceHost builder.Step
+}
+
+var Steps = steps{
+	Publisher: builder.NewStep(
+		"publisher/s2i",
```

Comment:
maybe we should find a better naming convention for steps IDs like `s2i/publisher` to reduce the risk of overriding steps f

Reply:
@astefanutti wondering if you have time to fix this too before merging

Reply:
Yes, better doing it while we are at it.

Reply:
The step IDs are now generated from the structs and automatically set on each step with reflection.